connectors/protocol/connectors.py

```diff
@@ -738,24 +743,43 @@ async def sync_starts(self, job_type):
             msg = f"Unknown job type: {job_type}"
             raise ValueError(msg)
 
-        doc = {
-            "status": Status.CONNECTED.value,
-            "error": None,
-        } | last_sync_information
+        if self.index.feature_use_connectors_api:
+            await self.index.api.connector_update_last_sync_info(
+                connector_id=self.id, last_sync_info=last_sync_information
+            )
+            await self.index.api.connector_update_status(
+                connector_id=self.id, status=Status.CONNECTED.value
+            )
+            await self.index.api.connector_update_error(
+                connector_id=self.id, error=None
+            )
```

Comment:
That's a little concerning - we do 3 calls to do single thing? Should we merge them into one call?

Reply:
Agreed, uniting status update + error update in one endpoint and leaving `update_last_sync_info` endpoint separately is a good way forward